### PR TITLE
Better modals & routing

### PR DESCRIPTION
**Pull request by PatrickShaw**
- Reusable modal stuff has been separated into its own Component
- Modal no longer uses `calc(...)` and `top`, etc which I thought was a little unnecessary when we've got `flexbox`.
- Routing makes a little more sense now imo

### CODE REVIEW

File: src/view/Overview.tsx [hunk 1]
```diff
@@ -21,8 +21,8 @@ export const Overview: SFC<{}> = observer(({ history }) => (
         history.push(`/specifications/${specification.id}`)
       }
       // Opens the 'Add Specification' modal
-      onAddSpecificationModalOpened={() => history.push('/add')}
+      onAddSpecificationModalOpened={() => history.push(`${match.url}/add`)}
```
Comment (PatrickShaw) [inline src/view/Overview.tsx]: Still couples the modal with the overview page but I think that's a problem to solve for another week.

File: src/state/SpecificationState.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ class BasicSpecificationState implements SpecificationState {
     title,
     path,
     description
-  }: AddedSpecification): Promise<boolean> {
+  }: AddedSpecification): Promise<void> {
```
Comment (calebj0seph) [inline src/state/SpecificationState.tsx]: So in future this method would throw an exception if the server returned an error?

File: src/view/AddSpecificationModal.tsx [hunk 3]
```diff
@@ -88,7 +81,7 @@ export class AddSpecificationModal extends Component<RouteComponentProps<{}>, {}
   @observable private showErrorModal: boolean = false;
 
   closeModal() {
-    this.props.history.replace('/');
+    this.props.history.goBack();
```
Comment (calebj0seph) [inline src/view/AddSpecificationModal.tsx]: How does this work if there is nothing in the history stack?

File: src/state/SpecificationState.tsx (hunk 2 above)
Comment (PatrickShaw) [reply src/state/SpecificationState.tsx]: Yes, didn't really give it much thought but i'd rather just make use of the already existing fetch exceptions else where in the code (maybe)

**Pull request merged**